readhomer_atlas/library/importers.py

```diff
@@ -88,23 +113,67 @@ def generate_branch(self, line):
                 self.nodes[data["urn"]] = node
 
     def apply(self):
-        full_content_path = os.path.join(
-            LIBRARY_DATA_PATH, self.version_data["content_path"]
-        )
+        full_content_path = self.library.versions[self.urn]["path"]
         with open(full_content_path, "r") as f:
             for line in f:
                 self.generate_branch(line)
-
         created_count = Node.objects.get(
             urn=self.version_data["urn"]
         ).get_descendant_count()
         print(f"{self.name}: {created_count + 1} nodes.", file=sys.stderr)
 
 
+def resolve_library():
```

Comment:
Might be able to simplify this logic quite a lot using `os.walk` instead of `os.listdir`?

Reply:
Thoughts on the new `LibraryDataResolver` class?  I separated it from the `Library` class so we don't have to crawl directories when running tests

Reply:
definite improvement!